ls1intum/Artemis
# `Development`: Update documentation for Jenkins pipeline templates

### Checklist
#### General



### Motivation and Context
Closes #6783.

### Description
With [release 6.3.0](https://github.com/ls1intum/Artemis/releases/tag/6.3.0) the structure of the Jenkins pipelines changed from declarative `Jenkinsfile`s to scripted `pipeline.groovy` pipelines.
This change was not properly reflected in the documentation for the Artemis server setup.

### Steps for Testing
Look at the updated documentation [here](https://artemis-platform--6958.org.readthedocs.build/en/6958/dev/setup.html#templates) and make sure it is understandable, there are no typos, and the formatting is correct.

### Reviews
